### ISSUE TITLE
warning: unused return value of `std::mem::replace` that must be used

### ISSUE THREAD

**Issue by SuperSandro2000** [pemistahl/grex#27]
When I compile grex with rust 1.44.1 on nix I get the following warning:

```
warning: unused return value of `std::mem::replace` that must be used
  --> src/regexp/regexp.rs:47:9
   |
47 | /         std::mem::replace(
48 | |             test_cases,
49 | |             test_cases.iter().map(|it| it.to_lowercase()).collect_vec(),
50 | |         );
   | |__________^
   |
   = note: `#[warn(unused_must_use)]` on by default
   = note: if you don't need the old value, you can just assign the new value directly

warning: 1 warning emitted
```

**Comment by pemistahl** [comment]
This is nothing to worry about. It's a warning, not an error. The return value of the function is not needed, so I don't save it. The warning message could be suppressed in the future to avoid confusion.

**Comment by SuperSandro2000** [comment]
Not sure why you closed the issue. The warning still exists and it will probably raise questions when I submit it to nixos.

**Comment by pemistahl** [comment]
The warning did not exist when I compiled the release with the back then current version of the Rust compiler. Just ignore it. It is irrelevant and `grex` will work just fine. I will fix the warning in version 1.2 but this is not worth a bug fix release as this is simply not a bug. If you want to submit `grex` to some Linux distribution then please simply explain the issue in the unlikely case that questions may arise. Thank you.
